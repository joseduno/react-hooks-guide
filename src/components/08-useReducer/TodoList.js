import React from 'react'
import TodoListItem from './TodoListItem'

const TodoList = ({todos, handleDone, handleDelete}) => {

    return (
        <ul className='list-group list-group-flush'>
            {todos.map((todo, i) => (
                <TodoListItem
                    key={todo.id}
                    todo={todo}
                    i={i}
                    handleDone={handleDone}
                    handleDelete={handleDelete}
                />
            ))}
        </ul>
    )
}

export default TodoList
