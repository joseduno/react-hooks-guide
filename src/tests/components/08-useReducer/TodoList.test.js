import { demoTodos } from "../../fixtures/demoTodos"
import { shallow } from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoList";

describe('Pruebas en <TodoList />', () => {

    const handleDone = jest.fn()
    const handleDelete = jest.fn()

    const wrapper = shallow(<TodoList
        todos={demoTodos}
        handleDone = {handleDone}
        handleDelete = {handleDelete}
    />)

    test('debe mostrarse el componente correctamente con la cantidad correcta de TodoListItem', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length)
    })

    test('debe tener las propiedades definidas de forma correcta', () => {
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function))
        expect(wrapper.find('TodoListItem').at(0).prop('handleDone')).toEqual(expect.any(Function))
    })
    
    
})
