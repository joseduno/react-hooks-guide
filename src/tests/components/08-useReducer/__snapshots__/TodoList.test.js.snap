// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> debe mostrarse el componente correctamente con la cantidad correcta de TodoListItem 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleDone={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleDone={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
